### PR TITLE
Add instructions for executing the AWS CodeBuild workflows locally

### PR DESCRIPTION
**Pull request by bmoffatt**
*Description of changes:*

change separated out from #159 

Added instructions to CONTRIBUTING.md for guidance on executing the CodeBuild testing locally based on https://docs.aws.amazon.com/codebuild/latest/userguide/use-codebuild-agent.html

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -43,6 +43,24 @@ GitHub provides additional document on [forking a repository](https://help.githu
 ## Finding contributions to work on
 Looking at the existing issues is a great way to find something to contribute on. As our projects, by default, use the default GitHub issue labels (enhancement/bug/duplicate/help wanted/invalid/question/wontfix), looking at any ['help wanted'](https://github.com/awslabs/aws-lambda-cpp-runtime/labels/help%20wanted) issues is a great place to start. 
 
+## Running the Integration Tests Locally
+
+The integration testing for the project creates, invokes, and deletes, Lambda functions. 
+These tests typically run  AWS CodeBuild, but may also be executed locally
```
Comment (marcomagdy) [inline CONTRIBUTING.md:49]:
```suggestion
These tests typically run on AWS CodeBuild, but may also be executed locally
```

So this literally runs CodeBuild locally using CodeBuild's docker image, right?

File: ci/codebuild_build.sh [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    docker_command+=" -e \"INITIATOR=$USERNAME\""
+else
+    docker_command+=" -e \"INITIATOR=$USER\""
+fi
+
+if [ -n "$local_agent_image" ]
+then
+    docker_command+=" $local_agent_image"
+else
+    docker_command+=" public.ecr.aws/codebuild/local-builds:latest"
```
Comment (marcomagdy) [inline ci/codebuild_build.sh:189]: pretty cool

**Review by marcomagdy — APPROVED** [review]
Amazing stuff.

File: ci/codebuild_build.sh (hunk 2 above)
Comment (bmoffatt) [reply ci/codebuild_build.sh:189]: yeah I love this script the codebuild folk wrote! My only complaint is that it doesn't auto detect aarch64, gotta pass in override image from their docs explicitly

File: CONTRIBUTING.md (hunk 1 above)
Comment (bmoffatt) [reply CONTRIBUTING.md:49]: yup!

**Pull request merged**